test/unittest/tests/block/test_extent_list_search.cpp

```diff
@@ -0,0 +1,296 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [extent_list2]: block_ext.c
+ * Test extent list functions part 3. (More to come.)
+ *
+ * Test extent list search functions: __block_off_srch_pair, and __block_off_match.
+ */
+
+#include <algorithm>
+#include <catch2/catch.hpp>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+#include "test_util.h"
+#include "../utils.h"
+#include "../utils_extlist.h"
+
+using namespace utils;
+
+/*!
+ * A test (off) and expected values (before and after) for __block_srch_pair.
+ */
+struct search_before_after {
+    wt_off_t off;
+    off_size *before;
+    off_size *after;
+
+    search_before_after(wt_off_t off, off_size *before, off_size *after)
+        : off(off), before(before), after(after)
+    {
+    }
+};
+
+TEST_CASE("Extent Lists: block_off_srch_pair", "[extent_list2]")
+{
+    /* Build Mock session, this will automatically create a mock connection. */
+    std::shared_ptr<MockSession> mock_session = MockSession::buildTestMockSession();
+    WT_SESSION_IMPL *session = mock_session->getWtSessionImpl();
+
+    std::vector<WT_EXT **> stack(WT_SKIP_MAXDEPTH, nullptr);
+
+    SECTION("search an empty list")
+    {
+        BREAK;
+        /* Offsets to search for. */
+        std::vector<wt_off_t> test_list{0, 4096, 3 * 4096}; // 3, 4096, 12,288.
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+
+        /* Test. */
+        WT_EXT dummy;
+        for (const wt_off_t &test : test_list) {
+            INFO("Search: off " << test);
+            /* Set to an invalid value to determine whether __block_off_srch_pair changed them. */
+            WT_EXT *before = &dummy;
+            WT_EXT *after = &dummy;
+            /* Call. */
+            __ut_block_off_srch_pair(&extlist, test, &before, &after);
+            /* Verify: All should be not found. */
+            REQUIRE(before == nullptr);
+            REQUIRE(after == nullptr);
+        }
+    }
+
+    SECTION("search a non-empty list")
+    {
+        BREAK;
+        /* Extents to insert to create an extent list to search. */
+        std::vector<off_size> insert_list{
```

Comment:
It is not sorted because sorting the input does not test whether __block_off_insert() sorts the input values. It is a better test to use unsorted.  Another reason is I cut and pasted it here from __block_off_insert(). It was easier to not sort it. I don't think it matters either way.

I do agree that it is not necessary to test __block_off_insert() here since it is done elsewhere.

Reply:
Right, i see what you mean here. My personal belief is that we should make it sorted, as it helps with the readability of the test. Since `__block_off_insert` is already tested in another function I don't see a need to make it complicated here.

Reply:
Done_